doc/ipython-notebooks/gaussian_process/variational_classifier.ipynb

```diff
@@ -629,14 +733,380 @@
+      "             elif idx==1:\n",
+      "                fit_condition=\"right-fit\"\n",
+      "             else:\n",
+      "                fit_condition=\"under-fit\"                           \n",
+      "             plots[idx][choice].set_title(\"Decision boundary (blue) \\n of %s on %s points (%s)\"%(name, type_name,fit_condition))\n",
+      "\n",
+      "    plt.show()"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "inf='../../../data/toy/banana.dat'\n",
+      "plot(inf)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "heading",
+     "level": 2,
+     "metadata": {},
+     "source": [
+      "<a id=\"largescale\">Inference Methods for sparse GPC Models</a> <a href='#toc'>[TOP]</a>"
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "TO DO: Theory of large-scale inference methods for sparse Gaussian processes will be discussed in a new notebook.\n",
+      "\n",
+      "TO DO: Inference methods for sparse GPC models (idea: stochastic gradient) (ongoing)\n",
+      "\n",
+      "TO DO: Offer an option to automatically learn K good inducing points given the number of inducing points, K, is specific."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "def train2(inference, linesearch, likelihood, x_train, y_train, x_inducing, kernel_log_sigma, kernel_log_scale):\n",
+      "    \n",
+      "    \"\"\"\n",
+      "    This function is used to train a GP classifer given an inference method\n",
+      "        \n",
+      "    Parameters:\n",
+      "        inference - an inference methods used to train the classifer\n",
+      "        linesearch - a linesearch method used in the inference method \n",
+      "        likelihood - likelihood function to model data\n",
+      "        x_train, x_test - features for training and testing\n",
+      "        y_train - labels for training\n",
+      "        kernel_log_sigma, kernel_log_scale hyper-parameters of covariance function    \n",
+      "    Returns:\n",
+      "        trained GPC model, name of inference method\n",
+      "    \"\"\"\n",
+      "\n",
+      "    mean_func = ZeroMean()\n",
+      "    kernel_sigma = 2*exp(2*kernel_log_sigma);\n",
+      "    kernel_func = GaussianKernel(10, kernel_sigma)\n",
+      "\n",
+      "    #Y is a sample-by-1 vector\n",
+      "    labels_train = BinaryLabels(y_train)\n",
+      "    #X is a feature-by-sample matrix\n",
+      "    features_train=RealFeatures(x_train)\n",
+      "    features_inducing=RealFeatures(x_inducing)\n",
+      "    \n",
+      "    inf = inference(kernel_func, features_train, mean_func, labels_train, likelihood, features_inducing)\n",
+      "    inf.set_scale(exp(kernel_log_scale))\n",
+      "    \n",
+      "    try:\n",
```

Comment:
A comment here helps people understand what is going on. You should add something